topdown/eval.go

```diff
@@ -403,7 +406,8 @@ func (e *eval) evalStep(iter evalIterator) error {
 			})
 		}
 	case *ast.Term:
-		rterm := e.generateVar(fmt.Sprintf("term_%d_%d", e.queryID, e.index))
+		// generateVar inlined here to avoid extra allocations in hot path
+		rterm := ast.VarTerm(fmt.Sprintf("%s_term_%d_%d", e.genvarprefix, e.queryID, e.index))
```

Comment:
Interesting that the compiler couldn't inline that itself 🤔

Reply:
Not sure if it was possible for the compiler to inline, but it’s a little more elaborate, as our manual inlining also reduces 2 sprintf calls to one